### PR TITLE
feat(controller): made sure controller is not divulging sensitive data when not in evaluate mode [ch5020]

### PR DESCRIPTION
**Pull request by lakatosandrei**

### CODE REVIEW

**Comment by edeustace** [comment]
tests?

File: packages/match/controller/src/__tests__/index.test.jsx [hunk 1]
```diff
@@ -110,17 +123,29 @@ describe('model', () => {
       });
     });
 
+    it('returns a new object, without using the same reference', () => {
```
Comment (edeustace) [inline packages/match/controller/src/__tests__/index.test.jsx]: object references aren't really an issue, when run in a secure environment, the `model` function will run remotely and be returns as json - so they'll never be the same reference.

File: packages/match/controller/src/__tests__/index.test.jsx [hunk 2]
```diff
@@ -131,6 +156,10 @@ describe('model', () => {
       result = await model(question, session, env);
     });
 
+    it('returns a new object, without using the same reference', () => {
```
Comment (edeustace) [inline packages/match/controller/src/__tests__/index.test.jsx]: see above

File: packages/match/controller/src/__tests__/index.test.jsx (hunk 1 above)
Comment (lakatosandrei) [reply packages/match/controller/src/__tests__/index.test.jsx]: If I didn't `clone` the object the model would never come back with rows that have values. So I had to clone it, look below at my comment.

File: packages/match/controller/src/index.js [hunk 3]
```diff
@@ -168,6 +170,10 @@ export function model(question, session, env, updateSession) {
 
     normalizedQuestion.rows.forEach(row => {
       correctResponse[row.id] = row.values;
+
```
Comment (lakatosandrei) [inline packages/match/controller/src/index.js]: @edeustace 
Because of the below statement I had to clone it.

**Pull request merged**